### PR TITLE
fix(matrix): fix matrix scroll bug

### PR DESCRIPTION
**Pull request by tobiaskneubuehler**

### CODE REVIEW

File: www/src/js/components/matrix/matrix.jsx [hunk 1]
```diff
@@ -4,123 +4,172 @@ import Resistance from '../matrixResistance/resistance';
...
-	}
+    @observable svg;
+
+    /**
+     * Scroll to matrix to it's top.
+     * We have to use «document.getElementsByClassName('main__matrix')» as the matrix container
+     * is not part of react (is not a component).
+     */
+    scrollMatrixToTop() {
+        const mainMatrixContainer = document.getElementsByClassName('main__matrix');
```
Comment (fxstr) [inline www/src/js/components/matrix/matrix.jsx]: Nope. https://reactjs.org/docs/refs-and-the-dom.html

Comment (fxstr) [reply www/src/js/components/matrix/matrix.jsx]: And also no to selecting an element from DOM every time you scroll up. The element should be a class property that is set when it's rendered.

File: www/src/js/components/matrix/matrix.jsx [hunk 2]
```diff
@@ -4,123 +4,172 @@ import Resistance from '../matrixResistance/resistance';
...
+        if (mainMatrixContainer[0]) {
```
Comment (fxstr) [inline www/src/js/components/matrix/matrix.jsx]: You can do it this way, I'd check the length of mainMatrixContainer though.

File: www/src/js/components/matrix/matrix.jsx (hunk 1 above)
Comment (tobiaskneubuehler) [reply www/src/js/components/matrix/matrix.jsx]: I sure know about react-refs, but the container is not in or a react-component, so we can not «ref it»: https://github.com/infect-org/frontend/blob/fix/matrix-scroll/www/src/html/index.html#L129

I think this is why you also use a query-selector in the matrixHeader: https://github.com/infect-org/frontend/blob/3ca7011fcdfb1a5a9756fd7b767c0cff3c4be0a9/www/src/js/components/matrix/matrixHeader.jsx#L16

should we move the div into a component?

Comment (tobiaskneubuehler) [reply www/src/js/components/matrix/matrix.jsx]: > And also no to selecting an element from DOM every time you scroll up. The element should be a class property that is set when it's rendered.

👍

Comment (fxstr) [reply www/src/js/components/matrix/matrix.jsx]: 🙈 you're right, indeed! Sorry about that!

**Pull request closed without merging**